src/main/scala/mesosphere/marathon/raml/VolumeConversion.scala

```diff
@@ -140,8 +142,10 @@ trait VolumeConversion extends ConstraintConversion with DefaultConversions {
   }
 
   implicit val volumePersistentReads: Reads[AppPersistentVolume, state.VolumeWithMount[Volume]] = Reads { volumeRaml =>
+    // If a disk profile is set, only DiskType.Mount is supported, while DiskType defaults to Root
+    val diskType = volumeRaml.persistent.profileName.map(_ => DiskType.Mount).getOrElse(volumeRaml.persistent.`type`.fromRaml)
```

Comment:
This could be viewed a DRY violation, but to me creating a function to return the "correct" default didn't look any better. I'd rather live with this being computed in both places, but I don't have a strong opinion.

Reply:
I'm rather under the opinion that we should:

1. Change the default to mount if no volume type is specified, and the profile is specified
2. (May be?) Add a validation error if a volume type other than mount is specified, and the profile is specified

I say "maybe" for the second item because it would be a breaking change, and we would potentially need a way to disable said breaking change if it puts their cluster in a bad state. This really to me feels like the right option here.

Reply:
IMO the second change is okay, even if it's a breaking change. Currenlty, DSS has only been available as beta and its update story not very good. Breakages were to be expected. Once DSS is GA, this will of course be different.

Reply:
If I'm not mistaken anyone _could_ set profiles to disk resources in Mesos, and use it in some way in a framework, correct? So, just because there's a profileName set doesn't guarantee that DSS was used to set it. Therefore, I would not want to fail validation. And also not b/c of the breaking change.

> Change the default to mount if no volume type is specified, and the profile is specified

@timcharper, this is what the PR does. I assume what you mean is it would be nicer to have an unmarshaller that takes the whole JSON object and defaults to mount if profile is set, and root if no profile is set? Right now, `type` has its own marshaller, which doesn't have the volume in scope. Meaning, the `fromRaml` only unmarshals the string defining disk type.

Reply:
@timcharper I understood what you meant – the original change applied `Mount` disregards what was configured. I spend some time refactoring this and added extensive tests for the diskType setting for both apps and pods. PTAL.